src/Symfony/Component/Messenger/Command/ConsumeMessagesCommand.php

```diff
@@ -177,10 +187,10 @@ protected function execute(InputInterface $input, OutputInterface $output)
             $retryStrategies[$receiverName] = null !== $this->retryStrategyLocator ? $this->retryStrategyLocator->get($receiverName) : null;
         }
 
-        if (null !== $input->getOption('bus')) {
-            $bus = $this->busLocator->get($input->getOption('bus'));
-        } else {
-            $bus = new RoutableMessageBus($this->busLocator);
+        if (null !== $busName = $input->getOption('bus')) {
+            $bus = $this->routableBus ? $this->routableBus->getMessageBus($busName) : $this->busLocator->get($busName);
+        } elseif (!$bus = $this->routableBus) {
+            $bus = $this->routableBus ? $this->routableBus : new RoutableMessageBus($this->routableBus);
```

Comment:
The condition `$this->routableBus` cannot be true since it's already part of the `elseif`

Reply:
Also shouldn't it be `new RoutableMessageBus($this->busLocator)`?

Reply:
fixed and covered by tests, thanks